.github/workflows/java-ec2-asg-retry.yml

```diff
@@ -0,0 +1,57 @@
+## Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+## SPDX-License-Identifier: Apache-2.0
+
+# This is a reusable workflow for running the Enablement test for App Signals.
+# It is meant to be called from another workflow.
+# Read more about reusable workflows: https://docs.github.com/en/actions/using-workflows/reusing-workflows#overview
+name: Java EC2 asg Retry
+on:
+  workflow_call:
+    inputs:
+      aws-region:
+        required: true
+        type: string
+      caller-workflow-name:
+        required: true
+        type: string
+
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  java-ec2-asg-attempt-1:
+    uses: ./.github/workflows/java-ec2-asg-test.yml
+    secrets: inherit
+    with:
+      aws-region: ${{ inputs.aws-region }}
+      caller-workflow-name: ${{ inputs.caller-workflow-name }}
+
+  java-ec2-asg-attempt-2:
+    needs: [ java-ec2-asg-attempt-1 ]
+    if: ${{ needs.java-ec2-asg-attempt-1.outputs.job-started != 'true' }}
+    uses: ./.github/workflows/java-ec2-asg-test.yml
+    secrets: inherit
+    with:
+      aws-region: ${{ inputs.aws-region }}
+      caller-workflow-name: ${{ inputs.caller-workflow-name }}
+
+  publish-metric-attempt-1:
+    needs: [ java-ec2-asg-attempt-1, java-ec2-asg-attempt-2 ]
+    if: always()
+    uses: ./.github/workflows/enablement-test-publish-result.yml
+    secrets: inherit
+    with:
+      aws-region: ${{ inputs.aws-region }}
+      caller-workflow-name: ${{ inputs.caller-workflow-name }}
+      validation-result: ${{ needs.java-ec2-asg-attempt-1.outputs.validation-result || needs.java-ec2-asg-attempt-2.outputs.validation-result }}
```

Comment:
Need some proof in PR description this works in all cases

Reply:
Ran test to verify this. added extra comment in PR description